x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/search/action/TransportQuerySearchApplicationAction.java

```diff
@@ -102,8 +103,17 @@ private static Map<String, Object> mergeTemplateParams(QuerySearchApplicationAct
         return mergedTemplateParams;
     }
 
-    private SearchSourceBuilder renderTemplate(Script script, Map<String, Object> templateParams) throws IOException {
-        TemplateScript compiledTemplate = scriptService.compile(script, TemplateScript.CONTEXT).newInstance(templateParams);
+    private SearchSourceBuilder renderTemplate(SearchApplication searchApplication, QuerySearchApplicationAction.Request request)
+        throws IOException, ValidationException {
+
+        final SearchApplicationTemplate template = searchApplication.searchApplicationTemplate();
+        final Map<String, Object> queryParams = request.queryParams();
+        final Script script = template.script();
+
+        template.validateTemplateParams(queryParams);
```

Comment:
Maybe not for this PR, but I think it would make sense to validate default params when the template is added or updated in a search application, otherwise I think we're good not to do validation on every request. WDYT?

Reply:
> I think it would make sense to validate default params when the template is added or updated in a search application

I think we can trust the admin / whoever has permissions for changing the template not to screw up on default parameters, and thus do not add unneeded validation to params that might not be changed by the user. But I'm more than happy to revisit this in a follow up PR.